lib/core/util.js

```diff
@@ -436,6 +436,47 @@ function parseHeaders (headers, obj) {
   return obj
 }
 
+/**
+ * @param {Record<string, string | string[]>} headers
+ * @returns {Buffer[]}
+ */
+function encodeHeaders (headers) {
+  const headerNames = Object.keys(headers)
+
+  /**
+   * @type {Buffer[]}
+   */
+  const rawHeaders = new Array(headerNames.length * 2)
+
+  let rawHeadersIndex = 0
+  for (const header of headerNames) {
+    let rawValue
+    const value = headers[header]
+    if (Array.isArray(value)) {
+      rawValue = new Array(value.length)
+
+      for (let i = 0; i < value.length; i++) {
+        rawValue[i] = Buffer.from(value[i])
+      }
+    } else {
+      rawValue = Buffer.from(value)
+    }
+
+    let headerBuffer = wellknownHeaderNameBuffers[header]
+    if (!headerBuffer) {
+      headerBuffer = Buffer.from(header)
+    }
```

Comment:
I don't think header name cache is necessary, it just increases memory usage.

Reply:
@mcollina wdyt?

Reply:
There are about 100 well-known headers. So it might not be a good idea to populate all the cache at once. 

What about a lazy initialized cache for a well-known header?